python/ray/tests/test_basic_2.py

```diff
@@ -558,6 +558,44 @@ def add(self, x):
     assert r1 == r2 == r3
 
 
+def test_actor_max_concurrency(ray_start_regular):
```

Comment:
Can we also test for the number of threads actually started? You could make a bunch of calls that return a thread ID (using https://docs.python.org/3/library/threading.html#threading.current_thread) and assert the set is the expected length. We should also do it for multiple callers (to catch the bug you identified).